lib/shared/puppet_self_service.rb

```diff
@@ -124,4 +124,153 @@ def self.filesystem_free(path)
     stat = Sys::Filesystem.stat(path)
     (stat.blocks_available.to_f / stat.blocks.to_f * 100).to_i
   end
+
+  # Execute a command line and return the result
+  #
+  # @param command_line [String] The command line to execute.
+  # @param timeout [Integer] Amount of time, in sections, allowed for
+  #   the command line to complete. A value of 0 will disable the timeout.
+  #
+  # @return [String] STDOUT from the command.
+  # @return [String] An empty string, if an exception is raised while
+  #   executing the command.
+  def self.exec_return_result(command_line, timeout = 300)
+    options = { timeout: timeout }
+    result = Exec.exec_cmd(command_line, **options)
+
+    if result.error.nil?
+      result.stdout
+    else
+      result.error
+    end
+  end
+
+  # Execute psql queries.
+  #
+  # @param sql [String] SQL to execute via a psql command.
+  # @param psql_options [String] list of options to pass to the psql command.
+  #
+  # @return (see #exec_return_result)
+
+  def self.psql_return_result(sql, psql_options = '')
+    command = %(su pe-postgres --shell /bin/bash --command "cd /tmp && #{PUP_PATHS[:server_bin]}/psql #{psql_options} --command \\"#{sql}\\"")
+    exec_return_result(command)
+  end
```

Comment:
As this PR already imports the `Exec` module, we should bring in the `psql` execution logic from `puppet_metrics_collector`:

https://github.com/puppetlabs/puppetlabs-puppet_metrics_collector/blob/v7.0.5/files/psql_metrics#L124-L159

That method eliminates quoting issues, sets a predictable time zone, and implements proper timeouts to prevent SQL transactions from being left open.